MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm

```diff
@@ -1299,6 +1299,11 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
     [mtlRendEnc setRenderPipelineState: cmdEncPool->getCmdClearMTLRenderPipelineState(_rpsKey)];
     [mtlRendEnc setDepthStencilState: cmdEncPool->getMTLDepthStencilState(isClearingDepth, isClearingStencil)];
     [mtlRendEnc setStencilReferenceValue: _mtlStencilValue];
+    [mtlRendEnc setCullMode:MTLCullModeNone];
+    [mtlRendEnc setTriangleFillMode:MTLTriangleFillModeFill];
+    [mtlRendEnc setDepthBias:0 slopeScale:0 clamp:0];
+    cmdEncoder->_viewportState.reset();
+    cmdEncoder->_scissorState.reset();
```

Comment:
This will cause MoltenVK to forget whatever viewports and scissors were set from the Vulkan API, if they were set beforehand. It won't even affect the current render encoder, since MoltenVK doesn't apply states until a draw call. You want to set the viewport and scissor state on the `MTLRenderCommandEncoder` instead, using `-[MTLRenderCommandEncoder setViewport:]` and `-[MTLRenderCommand setScissorRect:]`; then the `markDirty()` call will cause MoltenVK to restore the settings from the pipeline (or `vkCmdSetViewports()`/`vkCmdSetScissorRects()`) the next time a draw call is made.